src/Internal/Wallet/Cip30Mock.purs

```diff
@@ -136,7 +136,7 @@ mkCip30Mock pKey mSKey = do
         (unwrap keyWallet).selectCollateral coinsPerUtxoUnit
           maxCollateralInputs
           utxos
-          >>= liftMaybe (error "No UTxOs at address")
```

Comment:
Sorry for late comment, but what does UTxO folding mean? Are not UTxOs things persistent in blockchain and cannot be created on spot?

Reply:
We're folding `Maybe (Array TransactionUnspentOutput)` into `Array TransactionUnspentOutput`, which is equivalent to `maybe [] identity`. I did this because we fail later when trying to use the collateral array with a better error. Tbh it's not clear which failure cases end up in `Maybe` from selectCollateral

Reply:
Thanks!